controllers/user_character_controller.go

```diff
@@ -0,0 +1,54 @@
+package controllers
+
+import (
+	"encoding/json"
+
+	"net/http"
+
+	"github.com/ShinnosukeSuzuki/techtrain-mission-ca-tech-dojo-golang/api/middleware"
+	"github.com/ShinnosukeSuzuki/techtrain-mission-ca-tech-dojo-golang/controllers/services"
+	"github.com/ShinnosukeSuzuki/techtrain-mission-ca-tech-dojo-golang/models"
+)
+
+// UserCharacter用のコントローラ構造体
+type UserCharacterController struct {
+	service services.UserCharacterServicer
+}
+
+// コンストラクタ関数
+func NewUserCharacterController(s services.UserCharacterServicer) *UserCharacterController {
+	return &UserCharacterController{service: s}
+}
+
+// ハンドラーメソッドを定義
+
+// GET /character/list
+func (c *UserCharacterController) UserCharacterGetHandler(w http.ResponseWriter, r *http.Request) {
+	// GET以外のリクエストはエラー
+	if r.Method != http.MethodGet {
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		return
+	}
+
+	xToken, _ := r.Context().Value(middleware.TokenType{}).(string)
+
+	// xTokenを元に一致するユーザーのキャラクターを取得
+	characters, err := c.service.UserCharacterGetService(xToken)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if characters.Characters == nil {
```

Comment:
sliceの空かどうかと nil かどうかは別のチェックになってしまうので、配列の要素があるかないかを見たい場合は、 `len()` で評価するようにしましょう！

Ref: https://qiita.com/Yashy/items/1d4f277998866b186e19

```suggestion
	if len(characters.Characters) == 0 {
```

Reply:
https://github.com/ShinnosukeSuzuki/techtrain-mission-ca-tech-dojo-golang/pull/3/commits/990324a87e4700b0108226c3dfd817839ade89a4
上記コミットで対応しました。